packages/graphql/src/modules/user/domain/User.ts

```diff
@@ -1,49 +1,95 @@
-export type Providers = 'facebook' | 'google' | 'apple';
+namespace User {
+  type Providers = 'facebook' | 'google' | 'apple';
 
-export interface ProvidedBy {
-  providedBy: Providers;
-}
+  type ProvidedBy = Readonly<{
```

Comment:
Do we really need `ReadOnly`? If we focus on functional programming I think using `const` is enough to avoid mutability problems. I suggest turn on some kind of eslint rule so that we can alert reassignments in `let` vars.